File: fvc-frontend/src/services/arrangeOrderService.ts
import apiService from './api';
import type {
  ArrangeOrderData,
  ContentItem,
  SaveArrangeOrderRequest,
  RandomizeArrangeOrderRequest,
  ContentType,
} from '../types/arrange';

const arrangeOrderService = {
  getArrangeOrder: async (
    competitionId: string,
    contentType: ContentType
  ): Promise<ArrangeOrderData> => {
    const response = await apiService.get<ArrangeOrderData>(
      `/competitions/${competitionId}/arrange-order`,
      { contentType }
    );
    return response.data;
  },

  saveArrangeOrder: async (
    competitionId: string,
    request: SaveArrangeOrderRequest
  ): Promise<void> => {
    await apiService.post(
      `/competitions/${competitionId}/arrange-order`,
      request
    );
  },

  randomizeArrangeOrder: async (
    competitionId: string,
    request: RandomizeArrangeOrderRequest
  ): Promise<ArrangeOrderData> => {
    const response = await apiService.post<ArrangeOrderData>(
      `/competitions/${competitionId}/arrange-order/randomize`,
      request
    );
    // POST returns T directly, but check if it's wrapped
    return (response as any).data || response;
  },

  getContentItems: async (
    competitionId: string,
    contentType: ContentType
  ): Promise<ContentItem[]> => {
    const response = await apiService.get<ContentItem[]>(
      `/competitions/${competitionId}/arrange-order/content-items`,
      { contentType }
    );
    return response.data;
  },
};

export default arrangeOrderService;

File: fvc-api/src/main/java/sep490g65/fvcapi/service/ArrangeOrderService.java
package sep490g65.fvcapi.service;

import sep490g65.fvcapi.dto.request.RandomizeArrangeOrderRequest;
import sep490g65.fvcapi.dto.request.SaveArrangeOrderRequest;
import sep490g65.fvcapi.dto.response.ArrangeOrderResponse;
import sep490g65.fvcapi.dto.response.ContentItemResponse;
import sep490g65.fvcapi.enums.ContentType;

import java.util.List;

public interface ArrangeOrderService {
    
    /**
     * Get arranged order for a competition and content type
     */
    ArrangeOrderResponse getArrangeOrder(String competitionId, ContentType contentType);
    
    /**
     * Save or update arranged order
     */
    void saveArrangeOrder(SaveArrangeOrderRequest request);
    
    /**
     * Randomize arrange order from registrations
     */
    ArrangeOrderResponse randomizeFromRegistrations(RandomizeArrangeOrderRequest request);
    
    /**
     * Get available content items (fist items or music content) for a competition
     */
    List<ContentItemResponse> getContentItems(String competitionId, ContentType contentType);
}

